### PR TITLE
Implement commands for editing, deleting, viewing activities

### PR DESCRIPTION
**Pull request by AlWo223**

### CODE REVIEW

**Review by yicheng-toh — COMMENTED** [review]
LGTM! Code looks neat. Take note of the use of logger level.

File: src/main/java/athleticli/commands/activity/EditActivityCommand.java [hunk 1]
```diff
@@ -1,4 +1,51 @@
...
+        logger.log(java.util.logging.Level.INFO, "Editing activity at index " + index);
+        ActivityList activities = data.getActivities();
+        try {
+            activities.set(index-1, activity);
+            logger.log(java.util.logging.Level.INFO, "Activity at index " + index + "successfully edited");
+            return new String[]{Message.MESSAGE_ACTIVITY_UPDATED, activity.toString(),
+                    String.format(Message.MESSAGE_ACTIVITY_COUNT, activities.size())};
+        } catch (IndexOutOfBoundsException e) {
+            logger.log(java.util.logging.Level.WARNING, "Activity index out of bounds");
+            throw new AthletiException(Message.MESSAGE_ACTIVITY_INDEX_OUT_OF_BOUNCE);
```
Comment (DaDevChia) [inline src/main/java/athleticli/commands/activity/EditActivityCommand.java:48]:
```suggestion
            throw new AthletiException(Message.MESSAGE_ACTIVITY_INDEX_OUT_OF_BOUNDS);
```
Spelling error.

File: src/main/java/athleticli/commands/activity/DeleteActivityCommand.java [hunk 2]
```diff
@@ -1,4 +1,44 @@
...
+    @Override
+    public String[] execute(Data data) throws AthletiException {
+        ActivityList activities = data.getActivities();
+        try {
+            final Activity activity = activities.get(index-1);
+            activities.remove(activity);
+            return new String[]{Message.MESSAGE_ACTIVITY_DELETED, activity.toString(),
+                    String.format(Message.MESSAGE_ACTIVITY_COUNT, activities.size())};
+        } catch (IndexOutOfBoundsException e) {
+            throw new AthletiException(Message.MESSAGE_ACTIVITY_INDEX_OUT_OF_BOUNCE);
```
Comment (DaDevChia) [inline src/main/java/athleticli/commands/activity/DeleteActivityCommand.java:39]:
```suggestion
            throw new AthletiException(Message.MESSAGE_ACTIVITY_INDEX_OUT_OF_BOUNDS);
```
Spelling error.

File: src/main/java/athleticli/ui/Parser.java [hunk 3]
```diff
@@ -110,16 +222,19 @@ public static Command parseCommand(String rawUserInput) throws AthletiException
...
         final String caption = arguments.substring(0, durationIndex).trim();
-        final String duration = arguments.substring(durationIndex + 9, distanceIndex).trim();
-        final String distance = arguments.substring(distanceIndex + 9, datetimeIndex).trim();
-        final String datetime = arguments.substring(datetimeIndex + 9).trim();
+        final String duration =
+                arguments.substring(durationIndex + Parameter.DURATION_SEPARATOR.length(), distanceIndex).trim();
+        final String distance =
+                arguments.substring(distanceIndex + Parameter.DISTANCE_SEPARATOR.length(), datetimeIndex).trim();
+        final String datetime =
+                arguments.substring(datetimeIndex + Parameter.DATETIME_SEPARATOR.length()).trim();
```
Comment (DaDevChia) [inline src/main/java/athleticli/ui/Parser.java:237]: Great use of named constants for better maintainability and readability, also prevents bugs in the future when parameters are changed

File: src/main/java/athleticli/ui/CommandName.java [hunk 4]
```diff
@@ -15,12 +15,16 @@ public class CommandName {
     public static final String COMMAND_ACTIVITY = "activity";
     public static final String COMMAND_CYCLE = "cycle";
     public static final String COMMAND_SWIM = "swim";
```
Comment (DaDevChia) [inline src/main/java/athleticli/ui/CommandName.java:17]: One suggestion is to change activity keywords to `add-activity`, `add-cycle`, etc to standardize the commands across the board.

**Review by DaDevChia — APPROVED** [review]
Overall code quality is excellent, and also very technically impressive code is added as well.

**Pull request merged**